spec/minefield_spec.rb

```diff
@@ -0,0 +1,282 @@
+require 'minefield'
+require 'toy_minefield'
+
+describe Minefield do
+
+  let(:m){ Minefield.new }
+  let(:size25){ Minefield.new(25) }
+
+  describe '#new' do
+    describe 'takes a field size argument' do
+      it 'defaults to a 10 by 10 field' do
+        expect(m.size).to eq 10
+      end
+      it 'can take an integer as the size' do
+        size12 = Minefield.new(12)
+        expect(size12.size).to eq 12
+      end
+      it 'raises error if the argument is not an integer' do
+        expect do
+          Minefield.new("intermediate")
+        end.to raise_error
+      end
+    end
+
+    describe 'adds a certain number of mines' do
+      it 'adds 9 mines to the default field' do
+        expect(m.number_of_mines).to be 9
+      end
+      it 'adds 100 mines to a 25x25 field' do
+        expect(size25.number_of_mines).to be 100
+      end
+    end
+
+    describe 'verifies that the number of mines is correct' do
+      it 'counts 9 mines on the default field' do
+        placed_mines = m.field.flatten.count { |cell| cell.mine }
+        expect(placed_mines).to be 9
+      end
+      it 'counts 100 mines on a 25x25 field' do
+        placed_mines = size25.field.flatten.count { |cell| cell.mine }
+        expect(placed_mines).to be 100
+      end
+    end
+
+    describe 'counts adjacent mines for every cell' do
+      let(:t){ ToyMinefield.new }
+      it 'correctly labels neighbors of a single mine' do
+        t.field[0][0].place_mine
+        t.generate_adjacent_mine_counts
+        expect(t.field[0][0].adjacent_mines).to be 0
+        expect(t.field[0][1].adjacent_mines).to be 1
+        expect(t.field[0][2].adjacent_mines).to be 0
+        expect(t.field[1][0].adjacent_mines).to be 1
+        expect(t.field[1][1].adjacent_mines).to be 1
+        expect(t.field[1][2].adjacent_mines).to be 0
+        expect(t.field[2][0].adjacent_mines).to be 0
+        expect(t.field[2][1].adjacent_mines).to be 0
+        expect(t.field[2][2].adjacent_mines).to be 0
+      end
+      it 'correctly labels neighbors of two mines' do
+        t.field[0][0].place_mine
+        t.field[0][2].place_mine
+        t.generate_adjacent_mine_counts
+        expect(t.field[0][0].adjacent_mines).to be 0
+        expect(t.field[0][1].adjacent_mines).to be 2
+        expect(t.field[0][2].adjacent_mines).to be 0
+        expect(t.field[1][0].adjacent_mines).to be 1
+        expect(t.field[1][1].adjacent_mines).to be 2
+        expect(t.field[1][2].adjacent_mines).to be 1
+        expect(t.field[2][0].adjacent_mines).to be 0
+        expect(t.field[2][1].adjacent_mines).to be 0
+        expect(t.field[2][2].adjacent_mines).to be 0
+      end
+      it 'correctly labels neighbors of multiple mines' do
+        t.field[0][0].place_mine
+        t.field[0][2].place_mine
+        t.field[1][0].place_mine
+        t.field[2][0].place_mine
```

Comment:
This really is a great way to set up this test..... I think we tried a rougher workaround.